File: schedulling-api/src/main/java/com/scheduling/modules/availability/dto/AvailabilityDTO.java
package com.scheduling.modules.availability.dto;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;
import java.time.LocalTime;
import lombok.Data;

@Data
public class AvailabilityDTO {

  /**
   * Obrigatório quando {@code specificDate} não é informado (disponibilidade recorrente semanal).
   * Ignorado (recalculado a partir de {@code specificDate}) quando esta é avulsa.
   */
  @Min(0)
  @Max(6)
  private Integer dayOfWeek;

  /** Quando informado, esta disponibilidade vale só nesta data — não recorrente. */
  private LocalDate specificDate;

  @NotNull(message = "O horário de início é obrigatório")
  private LocalTime startTime;

  @NotNull(message = "O horário de término é obrigatório")
  private LocalTime endTime;

  private boolean active = true;
}

File: schedulling-ui/src/modules/availability/dtos/availability.dto.ts
export interface AvailabilityResponseDTO {
  id: string;
  dayOfWeek: number;
  specificDate?: string | null; // YYYY-MM-DD, presente quando é avulsa
  startTime: string; // HH:mm:ss
  endTime: string;
  active: boolean;
}

export interface AvailabilityDTO {
  dayOfWeek?: number;
  specificDate?: string;
  startTime: string;
  endTime: string;
  active: boolean;
}

export interface BlockAvailabilityDTO {
  startDateTime: string;
  endDateTime: string;
  reason?: string;
}
